compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp

```diff
@@ -375,11 +349,18 @@ LogicalResult AIETargetBackend::serializeExecutable(
     ParserConfig pcfg(variantOp->getContext());
     llvm::SourceMgr srcMgr;
 
-    OwningOpRef<ModuleOp> owningModuleOp =
-        parseSourceFile<ModuleOp>(inputMlirPath, srcMgr, pcfg);
+    // TODO(newling) check if we just just pass DeviceOp around directly,
+    // is is really necessary to wrap it in a module?
```

Comment:
Yea I'd prefer not to do that - not 100% sure but I don't think nested ops will survive if the ModuleOp itself is freed somewhere.

Reply:
I can get end-to-end numerical tests to run if I remove https://github.com/nod-ai/iree-amd-aie/blob/aa112f7d3bcb7dd990dd4f87b4a1f1ac468c911b/compiler/plugins/target/AMD-AIE/aie/AIEOps.td#L26 

I think I'll just do that. I don't see why aie.device must have a module as a parent. It's isolated from above, and doesn't seem to use the module parents symbol table for anything. 

In this PR or a follow-up, I don't mind

Reply:
Here is fine with me (you know I'm not big on sticking to "independent" changes :smile:)